### PR TITLE
Show docs on hover for keywords and primitives

### PR DESCRIPTION
**Pull request by Veykril**
lAWFadkziX

It's a bit annoying that this requires the `SyntaxNode` and `Semantics` to be pulled through `hover_for_definition` just so we can get the `std` crate but I couldn't think of a better way.

### CODE REVIEW

**Review by matklad — APPROVED** [review]
>It's a bit annoying that this requires the SyntaxNode and Semantics to be pulled through hover_for_definition just so we can get the std crate but I couldn't think of a better way.

:+1: for recognizing a strong code smell early. I think a general solution is applicable:

If an argument to the function feels awkward, replace the argument with the result of the computation involving this argument. In this case, I think we can add an `Option<FamousDefs>` argument instead of `Sema`, `SyntaxNode` pair.

File: Cargo.lock [hunk 1]
```diff
@@ -1926,18 +1926,18 @@ dependencies = [
...
  "xshell-macros",
 ]
 
 [[package]]
 name = "xshell-macros"
-version = "0.1.9"
+version = "0.1.8"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "6093c460064572007f885facc70bb0ca5e40a83ea7ff8b16c1abbee56fd2e767"
+checksum = "7f6af9f8119104697b0105989a73c578ce33f922d9d6f3dae0e8ae3d538db321"
```
[the remark this answers is not in the document]
Comment (Veykril) [reply Cargo.lock:1940]: Huh this wasn't there before, I think this got in when I rebased earlier, weird...

File: crates/ide/src/hover.rs [hunk 2]
```diff
@@ -272,6 +274,24 @@ fn hover_markup(
...
+    let markup = markup.as_str();
+    let markup = if !markdown {
+        remove_markdown(markup)
+    } else if links_in_hover {
+        rewrite_links(db, markup, &def)
+    } else {
+        remove_links(markup)
+    };
+    Markup::from(markup)
+}
```
Comment (matklad) [inline crates/ide/src/hover.rs:293]: I also feel that our docs pipeline is in the need of refactor. In particular, markdown removal really should be a methond on markap, called from an ide:

```rust
// in handlers.rs

if snap.config.caps.support_markdown() {
  markup.as_markdown()
} else {
  markup.as_plain_text()
}
```

bors r+

Comment (kjeremy) [reply crates/ide/src/hover.rs:293]: See #4326 and #6576

**Pull request merged**